### PR TITLE
Add Task 4 GET/api/articles

### PR DESCRIPTION
**Pull request by Mike777p**
Refactored Task 4

### CODE REVIEW

File: MessedUp.com/MessedUUpArticleModel.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+//  Function to return an array with all articles
+// exports.fetchArticles = ((error)=>{
+//     return db.query(`SELECT * FROM articles`).then((data)=>{
+//         return data.rows
+//     })
+// })
+
+exports.fetchCommentCounts = (error) => {
+    return db.query(`SELECT articles.*, COUNT(comments.article_id) AS comment_count
```
Comment (lewispricey) [inline MessedUp.com/MessedUUpArticleModel.js:11]: Great use of the count aggregate function here to get the number of comments in one sql query 👍

File: MessedUp.com/MessedUUpArticleModel.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    LEFT JOIN comments ON articles.article_id = comments.article_id
+    GROUP BY articles.article_id;`).then(data => {
+        // console.log(data.rows)
+      return data.rows;
+    })
+}
+
+// // Function return an object of a count of coments per article ID
+// exports.fetchCommentCounts = (error) => {
+//     return db.query(`SELECT article_id, COUNT(*) as count
```
Comment (lewispricey) [inline MessedUp.com/MessedUUpArticleModel.js:22]: Please remove any unrequired bits of code

File: MessedUp.com/messedUpArticleController.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            return { ...obj, comment_count: parseInt(obj.comment_count) }
+        })
+
+        finishedData.sort((a, b) => {
+            const dateA = new Date(a.created_at);
+            const dateB = new Date(b.created_at);
+            return dateB - dateA;
+          });
+
+        return response.status(200).json(finishedData);
```
Comment (lewispricey) [inline MessedUp.com/messedUpArticleController.js:18]: What key would we idealy send this information back on?

**Review by lewispricey — CHANGES_REQUESTED** [review]
Great work on this one so far Michael!

Really like the test you've written for this checking the length of the array before looking over to check the properties and the values.

I've left an individual comment on on the sort method you have inside of the controller, essentualy we don't want to do any data manipliation here and we can add an extra line the DB query to manage the sort order.

Overall awesome work, please have a look at using the database sort and send though another PR when you're ready!

**Comment by robert112233** [comment]
I might be wrong but I can't see the changes made that lewis suggested. There is still "sort" being used and data not being sent back on an object with a key. Let me know if you have changed stuff and I just can't see it

File: MessedUp.com/messedUpArticleController.js (hunk 3 above)
Comment (Mike777p) [reply MessedUp.com/messedUpArticleController.js:18]: is this msg?

Comment (Mike777p) [reply MessedUp.com/messedUpArticleController.js:18]: or you mean .send?

**Pull request merged**